rholang/src/main/scala/coop/rchain/rholang/interpreter/errors.scala

```diff
@@ -11,100 +11,87 @@ object errors {
   type InterpreterErrorsM[M[_]] = MonadError[M, InterpreterError]
   def interpreterErrorM[M[_]: Monad: InterpreterErrorsM] = MonadError[M, InterpreterError]
 
-  trait InterpreterError extends Throwable {
-    override def getMessage = toString
+  sealed abstract class InterpreterError(message: String) extends Throwable(message) {
+
+    def this(message: String, cause: Throwable) {
+      this(message)
+      initCause(cause)
```

Comment:
is `initCause` coming from `Throwable` itself?

Reply:
Yes